fragment.go

```diff
@@ -1708,14 +1750,15 @@ func (f *fragment) importValue(columnIDs, values []uint64, bitDepth uint, clear
 	if err := func() error {
 		for i := range columnIDs {
 			columnID, value := columnIDs[i], values[i]
-
 			var err error
 			if smallPath {
-				_, err = f.setValueBase(columnID, bitDepth, value, clear)
-			} else {
-				_, err = f.importSetValue(columnID, bitDepth, value, clear)
-			}
-			if err != nil {
+				ts, tc, err := f.positionsForValue(columnID, bitDepth, value, clear)
+				if err != nil {
+					return errors.Wrap(err, "getting positions for value")
+				}
+				toSet = append(toSet, ts...)
+				toClear = append(toClear, tc...)
```

Comment:
I don't have a really better idea yet, but it feels to me like this is a lot of allocation-and-appending going on, and it might make more sense to build this more like `append`, something like:

`toSet, toClear, err = f.positionsForValue(columnID, bitDepth, value, clear, toSet, toClear)`

I'd also suggest making them both slightly larger than 50% of the size -- as is, we can guess at least one of them will need to grow, even if we don't know which one.

Reply:
I went ahead and did 5/8ths for the initial alloc. I think you're right about making it append style... will do that next.